standalone/test/integration_tests/api/integrationtest_scanner_api.cpp

```diff
@@ -128,232 +164,163 @@ ScannerConfiguration ScannerAPITests::generateScannerConfig(const std::string& h
-{
-  auto stop_req_received_barrier = std::make_unique<util::Barrier>();
-
-  EXPECT_CALL(*scanner_mock_, receiveControlMsg(_, data_conversion_layer::stop_request::serialize()))
-      .WillOnce(OpenBarrier(stop_req_received_barrier.get()));
-  return stop_req_received_barrier;
-}
-
-std::unique_ptr<util::Barrier> ScannerAPITests::prepareMonitoringFrameBarrier(
-    const std::vector<data_conversion_layer::monitoring_frame::Message>& msgs)
+void ScannerAPITests::setUpScannerHwMock()
 {
-  const auto timestamp{ util::getCurrentTime() };
-  const auto scan{ createReferenceScan(msgs, timestamp) };
-
-  auto monitoring_frame_barrier = std::make_unique<util::Barrier>();
-
-  EXPECT_CALL(user_callbacks_,
-              LaserScanCallback(AllOf(ScanDataEqual(scan), TimestampInExpectedTimeframe(scan, timestamp))))
-      .WillOnce(OpenBarrier(monitoring_frame_barrier.get()));
-  return monitoring_frame_barrier;
+  hw_mock_.reset(new StrictMock<ScannerMock>{ HOST_IP_ADDRESS, port_holder_ });
 }
 
-void ScannerAPITests::startScanner()
-{
-  prepareScannerMockStartReply();
-  std::future<void> start_future;
-  EXPECT_DOES_NOT_BLOCK(start_future = scanner_->start(););
-  EXPECT_FUTURE_IS_READY(start_future) << "Scanner::start() not finished";
-}  // namespace psen_scan_v2_standalone_test
-
-void ScannerAPITests::stopScanner()
-{
-  auto stop_req_barrier = prepareStopRequestBarrier();
-
-  std::future<void> stop_future;
-  EXPECT_DOES_NOT_BLOCK(stop_future = scanner_->stop(););
-  EXPECT_TRUE(stop_req_barrier->waitTillRelease(DEFAULT_TIMEOUT)) << "Stop request not received";
-
-  scanner_mock_->sendStopReply();
-  EXPECT_FUTURE_IS_READY(stop_future) << "Scanner::stop() not finished";
-}
-
-void ScannerAPITests::sendMonitoringFrames(const std::vector<data_conversion_layer::monitoring_frame::Message>& msgs)
-{
-  for (const auto& msg : msgs)
-  {
-    scanner_mock_->sendMonitoringFrame(msg);
-    // Sleep to ensure that message are not sent too fast which might cause messages overwrite in socket buffer
-    std::this_thread::sleep_for(10ms);
-  }
-}
-
-TEST_F(ScannerAPITests, testStartFunctionality)
+TEST_F(ScannerAPITests, shouldSendStartRequestAndReturnValidFutureWhenLaunchingWithValidConfig)
 {
   setUpScannerConfig();
-  setUpScannerV2();
-  setUpScannerMock();
-  const auto start_req_received_barrier = prepareStartRequestBarrier(*config_);
+  setUpScannerV2Driver();
+  setUpScannerHwMock();
+  util::Barrier start_req_received_barrier;
+  EXPECT_START_REQUEST_CALL(*hw_mock_, *config_).WillOnce(OpenBarrier(&start_req_received_barrier));
 
   std::future<void> start_future;
-  EXPECT_DOES_NOT_BLOCK(start_future = scanner_->start(););
+  EXPECT_DOES_NOT_BLOCK(start_future = driver_->start(););
 
-  ASSERT_TRUE(start_req_received_barrier->waitTillRelease(DEFAULT_TIMEOUT)) << "Start request not received";
+  EXPECT_BARRIER_OPENS(start_req_received_barrier, DEFAULT_TIMEOUT) << "Start request not received";
```

Comment:
This used to be a ASSERT. Is it safe to switch to EXPECT?

Reply:
I don't see why it should be less safe here than with the other barriers. The start barriers were assert and the others expect. I understand that when the start fails the rest will probably fail as well, in reality but since we are using a mock that does not really care about the config we might get more (unrelated) errors afterwards that we can fix as well (which is to my knowledge the reason to use Expect at all)
If you think with multi threading this might cause trouble or the start is some special case that needs the assertion then we should consider switching to assert for all barrier checks. I don't have strong feelings about weather to use Expect or Assert.